tests/tracer/test_tracer.py

```diff
@@ -1706,3 +1708,30 @@ def test_tracer_api_version():
 
     t.configure(api_version="v0.4")
     assert isinstance(t.writer._encoder, MsgpackEncoderV03)
+
+
+@pytest.mark.parametrize("enabled", [True, False])
+def test_tracer_memory_leak(enabled):
+    spans = weakref.WeakSet()
+
+    # Filter to ensure we don't send the traces to the writer
+    class DropAllFilter:
+        def process_trace(self, trace):
+            return None
+
+    t = Tracer()
+    t.enabled = enabled
+    t.configure(settings={"FILTERS": [DropAllFilter()]})
+
+    # Run test in a function to ensure all local references to
+    # `span` are removed
```

Comment:
Doesn't setting `span = None` after the loop work the same? 🤔

Reply:
yeah, probably. I originally had `del span` which felt weird.

I can update to do `= None` that is nice too.